### PR TITLE
update code to be consist with new ros2 core

### PR DESCRIPTION
**Pull request by ahuizxc**
use "declare_parameter" instead of "get_parameter" and removed some compiler warnings.

### CODE REVIEW

**Comment by ahuizxc** [comment]
@clalancette clalancette  would you help to review this pr? thanks a lot.

File: src/DepthImageToLaserScanROS.cpp [hunk 1]
```diff
@@ -47,36 +47,29 @@
 using namespace depthimage_to_laserscan;
 
 DepthImageToLaserScanROS::DepthImageToLaserScanROS(rclcpp::Node::SharedPtr & node):node_(node) {
-
-  cam_info_sub_ = node_->create_subscription<sensor_msgs::msg::CameraInfo>("depth_camera_info",
+  auto qos = rclcpp::QoS(rclcpp::KeepLast(10));
```
Comment (clalancette) [inline src/DepthImageToLaserScanROS.cpp]: We may as well just get rid of this and use `rclcpp::SensorDataQoS()`, basically everywhere below.

File: src/DepthImageToLaserScanROS.cpp [hunk 2]
```diff
@@ -47,36 +47,29 @@
...
-  float scan_time = 0.033;
-  node_->get_parameter("scan_time", scan_time);
+  float scan_time = node_->declare_parameter("scan_time", 0.033);
   dtl_.set_scan_time(scan_time);
 
-  float range_min = 0.45;
-  float range_max = 10.0;
-  node_->get_parameter("range_min", range_min);
-  node_->get_parameter("range_max", range_max);
+  float range_min = node_->declare_parameter("range_min", 0.15);
```
Comment (clalancette) [inline src/DepthImageToLaserScanROS.cpp]: This used to be a default of 0.45 and 10.0, respectively.  Why are we changing the defaults?

File: src/DepthImageToLaserScanROS.cpp [hunk 3]
```diff
@@ -47,36 +47,29 @@
...
   dtl_.set_range_limits(range_min, range_max);
 
-  int scan_height = 1;
-  node_->get_parameter("scan_height", scan_height);
+  int scan_height = node_->declare_parameter("scan_height", 1);
   dtl_.set_scan_height(scan_height);
 
-  std::string output_frame = "camera_depth_frame";
-  node_->get_parameter("output_frame", output_frame);
+  std::string output_frame = node_->declare_parameter("output_frame", "camera_depth_optical_frame");
```
Comment (clalancette) [inline src/DepthImageToLaserScanROS.cpp]: Again, why are we changing the defaults?

File: src/DepthImageToLaserScanROS.cpp [hunk 4]
```diff
@@ -99,7 +92,7 @@ void DepthImageToLaserScanROS::depthCb(const sensor_msgs::msg::Image::SharedPtr
   try
   {
     sensor_msgs::msg::LaserScan::SharedPtr scan_msg = dtl_.convert_msg(image, cam_info_);
-    scan_pub_->publish(scan_msg);
+    scan_pub_->publish(*scan_msg);
```
Comment (clalancette) [inline src/DepthImageToLaserScanROS.cpp]: To make this compatible with zero-copy intra-process comms, we should change `convert_msg` to return a `std::unique_ptr`, and then publish by doing:

```
scan_pub_->publish(std::move(scan_msg));
```

**Review by clalancette — CHANGES_REQUESTED** [review]
Thanks for the PR!  I have a few things inline that we should do before getting this in.  Also, I've created a `dashing-devel` branch and re-targeted this PR to that.

File: src/DepthImageToLaserScanROS.cpp (hunk 2 above)
Comment (ahuizxc) [reply src/DepthImageToLaserScanROS.cpp]: My mistake:), I  changed this value before according to my demand during developing and forget to revert it back

**Comment by ahuizxc** [comment]
@clalancette hi clalancette i have already adapt code from your comment, and add a launch file and param.yaml, thanks for your review!

File: CHANGELOG.rst [hunk 5]
```diff
@@ -2,6 +2,11 @@
 Changelog for package depthimage_to_laserscan
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+2.2.1 (2019-03-29)
```
Comment (clalancette) [inline CHANGELOG.rst]: Let's not modify the CHANGELOG with this PR; that will get done during release time.

File: cfg/param.yaml [hunk 6]
```diff
@@ -0,0 +1,7 @@
+depthimage_to_laserscan:
+  ros__parameters:
+    scan_time: 0.033
+    range_min: 0.45
+    range_max: 10.0
+    scan_height: 1
+    output_frame: camera_depth_optical_frame
```
Comment (clalancette) [inline cfg/param.yaml]: Nit: newline at end of file.

File: launch/depthimage_to_laserscan.launch.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    param_config_dir = os.path.join(get_package_share_directory('depthimage_to_laserscan'), 'cfg', 'param.yaml')
+    return LaunchDescription([
+        Node(
+            package='depthimage_to_laserscan',
+            node_executable='depthimage_to_laserscan_node',
+            node_name='depthimage_to_laserscan_node',
+            remappings=[('depth','/camera/depth/image_rect_raw'),
+                        ('depth_camera_info', '/camera/depth/camera_info')],
+            parameters=[param_config_dir])
+    ])
```
Comment (clalancette) [inline launch/depthimage_to_laserscan.launch.py]: Nit: newline at end of file.

File: launch/depthimage_to_laserscan.launch.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# /* Author: Gary Liu */
+
+import os
+from ament_index_python.packages import get_package_share_directory
+from launch import LaunchDescription
+from launch.actions import DeclareLaunchArgument
```
Comment (clalancette) [inline launch/depthimage_to_laserscan.launch.py]: It doesn't look like this import is used, so please remove it.

File: launch/depthimage_to_laserscan.launch.py [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+from launch_ros.actions import Node
+from launch.substitutions import LaunchConfiguration
```
Comment (clalancette) [inline launch/depthimage_to_laserscan.launch.py]: It doesn't look like this import is used, so please remove it.

File: launch/depthimage_to_laserscan.launch.py [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+from launch.actions import IncludeLaunchDescription
```
Comment (clalancette) [inline launch/depthimage_to_laserscan.launch.py]: It doesn't look like this import is used, so please remove it.

File: launch/depthimage_to_laserscan.launch.py [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+from launch.launch_description_sources import PythonLaunchDescriptionSource
```
Comment (clalancette) [inline launch/depthimage_to_laserscan.launch.py]: It doesn't look like this import is used, so please remove it.

File: launch/depthimage_to_laserscan.launch.py [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+from launch.substitutions import ThisLaunchFileDir
```
Comment (clalancette) [inline launch/depthimage_to_laserscan.launch.py]: It doesn't look like this import is used, so please remove it.

**Review by clalancette — CHANGES_REQUESTED** [review]
Thanks for the updates!  I have a few more minor comments; once those are fixed, this should be ready to go in.

**Comment by ahuizxc** [comment]
All done!! @clalancette

**Review by clalancette — APPROVED** [review]
Looks good to me, thanks for iterating.

**Pull request merged**